core/tests/test_utils.py

```diff
@@ -3092,11 +2859,10 @@ def __init__(self, f, exception, num_tries_before_success):
         super(FailingFunction, self).__init__(f)
         self._exception = exception
         self._num_tries_before_success = num_tries_before_success
-        self._always_fail = (
-            self._num_tries_before_success == FailingFunction.INFINITY)
+        self._always_fail = num_tries_before_success == self.INFINITY
         self._times_called = 0
 
-        if not (self._num_tries_before_success >= 0 or self._always_fail):
+        if self._num_tries_before_success < 0 and not self._always_fail:
```

Comment:
I find the previous formulation easier to understand, especially in light of the error message.